main.nf

```diff
@@ -970,7 +973,65 @@ if (params.protocol != 'amplicon') {
     }
     ch_sort_bam_metrics = ch_ivar_metrics
     ch_sort_bam_varscan2 = ch_ivar_varscan2
-    ch_sort_bam_consensus = ch_ivar_consensus
+    ch_sort_bam_bcftools = ch_ivar_bcftools
+
+    process IVAR_VARIANTS {
+        tag "$sample"
+        label 'process_medium'
+        publishDir "${params.outdir}/variants/ivar/variants", mode: params.publish_dir_mode
+
+        when:
+        !params.skip_variants
+
+        input:
+        set val(sample), val(single_end), val(is_sra), file(bam) from ch_ivar_variants
+        file fasta from ch_fasta_ivar_variants.collect()
+        file gff from ch_gff_ivar_variants.collect().ifEmpty([])
+
+        output:
+        set val(sample), val(single_end), val(is_sra), file("*.tsv") into ch_ivar_variants_tsv
+
+        script:
+        features = params.gff ? "-g $gff" : ""
+        """
+        samtools mpileup \\
```

Comment:
As we need samtools mpileup for both variant and consensus for ivar, and we also need mpileup for varscan, wouldn't be more simple to have a process for mpileup, so the pipeline don't have to run it 3 times? It's a time consuming process.